base/math.jl

```diff
@@ -128,15 +128,17 @@ end
 @inline function exp2(x::Base.BitInteger)
     if x > 1023
         Inf64
-    elseif x < -1074
-        Float64(0.0)
     elseif x <= -1023
-        # Result will be a subnormal number
-        reinterpret(Float64, Int64(1) << (x + 1074))
+        if x < -1074
```

Comment:
Actually, you might not need this branch at all (if `x < -1074`, the bitshift should still give the correct answer).